autoendpoint/src/routers/webpush.rs

```diff
@@ -166,6 +167,13 @@ impl WebPushRouter {
             )
             .await
             .map_err(|e| ApiErrorKind::Router(RouterError::SaveDb(e)).into())
+            .map(|_| {
+                self.metrics
+                    .incr_with_tags("notification.message.stored")
+                    .with_tag("topic", &notification.headers.topic.is_some().to_string())
+                    // TODO: include `internal` if meta is set.
+                    .send();
+            })
```

Comment:
I kinda debated that a bit. Basically, the problem was the tag info, but perhaps I just need to pass that along as a parameter...

Reply:
In this specific case it looked like the only tag info needed (topic) was available in the object passed to `save_message` (it's a slightly different `Notification` type, so it ends up being `notification.topic.is_some()` instead)

Reply:
Also realized that part of my waffling on this was because I wanted to have a consistent layer where we record metrics. I think you're right that `DbClientImpl` is better than the calling layer, even if that means that the `DbClientImpl` may not have *all* the contexts that we may need or want. We should just add those to the `Notification` class when we need to.